### PR TITLE
Gp19 elasticsearch write

### PR DESCRIPTION
**Pull request by GBurley-Jones**
In this branch I have moved the main application into a new maven module so this is all the same code as previously. Only the Elasticsearch code needs reviewing

### CODE REVIEW

File: SpringProject/ElasticSearch/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+version: '2.0'
+
+services:
+  mysql:
+    image: mysql:5.7
```
Comment (ed-steele-rdm-els) [inline SpringProject/ElasticSearch/docker-compose.yml]: Why are you using an older version of mysql? 8.0 is the latest version

**Review by ed-steele-rdm-els — COMMENTED** [review]
I can't really review the functionality of this (there's a lot to unpack), but I left some vague style guideline comments

File: SpringProject/ElasticSearch/docker-compose.yml (hunk 1 above)
Comment (GBurley-Jones) [reply SpringProject/ElasticSearch/docker-compose.yml]: Good question, I all honesty I am not sure there is a reason! I have updated this to say mysql:8.0 and it still runs so I guess its okay?

**Comment by ed-steele-rdm-els** [comment]
> I can't really review the functionality of this (there's a lot to unpack), but I left some vague style guideline comments

Which github seems to have lost... 
basically; 

- var https://openjdk.java.net/projects/amber/LVTIstyle.html - don't overuse it. You should be writing the type of an object unless there is some good reason not to, e.g. `var someList = new ArrayList<String>();` 
- If created or deleting your indexes fail then the logs don't help find which index failed, and they'll still exist - look into having a retry policy or giving more info in the logs
- There's a bit when you load this json file based on where it is in the project - any reason for not putting that info into more standard config?

File: SpringProject/ElasticSearch/docker-compose.yml (hunk 1 above)
Comment (ed-steele-rdm-els) [reply SpringProject/ElasticSearch/docker-compose.yml]: Yes probably although I suspect Mendeley infra can't handle 8.0 so check where this is being built and deployed (if it is), if you're building everything from stratch it should be fine

Comment (GBurley-Jones) [reply SpringProject/ElasticSearch/docker-compose.yml]: We are building from scratch so I think it should be fine with 8.0 thanks!

**Review by Piotrek605 — APPROVED** [review]

**Pull request merged**